test/app.test.js

```diff
@@ -45,6 +45,22 @@ describe('Main Generator', function () {
     it('should generate a package.json file', function () {
       assert.file('temp/package.json');
     });
+
+    it('should have populated package.json correctly', function (done) {
+      fs.readFile('temp/package.json', function (err, data) {
```

Comment:
Does Yeoman not offer this.fs in tests as well? Also, their sample tests on http://yeoman.io/authoring/testing.html seem to be a little simpler than what I think you are trying to do.